### PR TITLE
[Avatar] Update colors

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris-internal/issues/1077

### WHY are these changes introduced?
Updates avatar background and text colors. 

|Before | After | 
|--|--|
|Screenshot 2023-11-30 at 9 18 59 AM|Screenshot 2023-11-30 at 9 13 37 AM|
Screenshot 2023-11-30 at 9 18 43 AM|Screenshot 2023-11-30 at 9 13 14 AM|

### How to 🎩

- 📕 [Storybook](https://5d559397bae39100201eedc1-nmmuuqskmo.chromatic.com/?path=/story/all-components-avatar--all)
- 👀 [Chromatic Diffs](https://shopify.chromatic.com/build?appId=5d559397bae39100201eedc1&number=21837)
- 🌀 [Spin](https://admin.web.avatar-uplift-fast-follows-bp4.laura-griffee.us.spin.dev)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] ~Updated the component's `README.md` with documentation changes~
- [ ] ~[Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide~

### CODE REVIEW

**Review by tjonx — APPROVED** [review]

**Review by annaborg — CHANGES_REQUESTED** [review]
Good on the color changes overall. Colors showing up correctly in the login screen, store switcher, search, org switcher, settings headers, orgs Stores page, orgs Users page. 


• Cannot test if showing up correctly in admin search. If there is above a certain number of Stores in the switcher within an Org admin in Plus, you are able to search for more stores that expand past the amount allowed in that switcher. 


• Not showing up on non-Plus Users and Permissions page
Screenshot 2023-11-30 at 12 41 42 PM
Screenshot 2023-11-30 at 12 42 55 PM


• All Plus users and permissions pages use the same color 
Screenshot 2023-11-30 at 12 47 12 PM
Screenshot 2023-11-30 at 12 47 26 PM
Screenshot 2023-11-30 at 12 47 43 PM


• Did we update the original avatars in account.shopify.com? It looks like they're using the new avatars in the login screen, but reverts back to old when you go to the different URL. I know account isn't react the last time I looked, so is that something we need to worry about right now? 
Screenshot 2023-11-30 at 12 34 32 PM


• No longer able to hover over avatars in the Store Settings header for neither Plus or Non-Plus to launch modal to edit and upload custom avatar. 


• Want to make sure that the above functionality persists, in addition to making sure that this PR doesn't cancel out an existing custom avatars that currently exist for Store navigation.

**Comment by lgriffee** [comment]
@annaborg 

> Good on the color changes overall. Colors showing up correctly in the login screen, store switcher, search, org switcher, settings headers, orgs Stores page, orgs Users page. 

Awesome! 🎉❤️✨ 

> Cannot test if showing up correctly in admin search. If there is above a certain number of Stores in the switcher within an Org admin in Plus, you are able to search for more stores that expand past the amount allowed in that switcher.

I've added more stores to the first business so you can test this now. From what I can tell it seems they are all displaying correctly!

> Not showing up on non-Plus Users and Permissions page 

You know I came across the same issue with Timeline recently (so this issue is currently one in production not something to do with this PR). And I couldn’t figure out why for the longest time but I finally tracked down what’s going on! There’s logic in the Avatar (from [this PR](https://github.com/Shopify/polaris/pull/7038)) that only applies the color styles when there is no source prop defined. So if there is a source provided but it fails to load (say an empty string gets passed in) then it just falls back to grey/initials. I've updated the logic so that if the image fails to load then it defaults back to the colored background instead. I've cut a new snapshot and confirmed the fix works. While loading the avatars are grey but as soon as they get initials they become colored. 

> All Plus users and permissions pages use the same color 

I investigated this and it looks like these are all the same since a blank string is being passed into the Avatar initials prop. I think the best solution for these instances is to add a name prop to that Avatar instance in the admin versus any changes in this PR. I will create a separate PR in the admin for this update.

> Did we update the original avatars in account.shopify.com? It looks like they're using the new avatars in the login screen, but reverts back to old when you go to the different URL. I know account isn't react the last time I looked, so is that something we need to worry about right now? 

Yeah since this is part of identity which is currently on the old rails version of Polaris we won't get updates for free here. That would need to be updated by that team.

> No longer able to hover over avatars in the Store Settings header for neither Plus or Non-Plus to launch modal to edit and upload custom avatar. Want to make sure that the above functionality persists, in addition to making sure that this PR doesn't cancel out an existing custom avatars that currently exist for Store navigation.

I spun up a fresh instance (without any of the changes here and with the latest admin changes) and couldn't get this functionality to work so I think this may be a spin limitation. This PR shouldn't effect that functionality.

File: .changeset/modern-plums-wonder.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': minor
+'@shopify/polaris-tokens': minor
+---
+
+Updated avatar background and text colors
```
Comment (chloerice) [inline .changeset/modern-plums-wonder.md]:
```suggestion
Updated `Avatar` background and text colors
```

**Review by chloerice — APPROVED** [review]

**Pull request merged**